samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -143,9 +233,26 @@ void handleContainerLaunchFail(String processorId, String containerId, String pr
    * @param resourceRequestState state of request in {@link ContainerAllocator}
    */
   @VisibleForTesting
-  void handleExpiredRequestWithHostAffinityEnabled(String processorId, String preferredHost,
+  void handleExpiredRequest(String processorId, String preferredHost,
       SamzaResourceRequest request, ContainerAllocator allocator, ResourceRequestState resourceRequestState) {
     boolean resourceAvailableOnAnyHost = allocator.hasAllocatedResource(ResourceRequestState.ANY_HOST);
+
+
+    // Case 1. Container placement actions can be taken in either cases of host affinity being set, in both cases
+    // mark the control action failed
+    if (hasActiveContainerPlacementAction(processorId)) {
+      resourceRequestState.cancelResourceRequest(request);
+      markContainerPlacementActionFailed(getPlacementActionMetadata(processorId).get(),
+          "failed the Control action because request for resources to ClusterManager expired");
+      return;
+    }
+
+    // Case 2. When host affinity is disabled wait for cluster resource manager return resources
+    if (!hostAffinityEnabled) {
+      return;
+    }
```

Comment:
Please add a comment to indicate that this is only to maintain parity with current-behavior, and TODO/SAMZA-XXXX: Enable any-host-retry for host-affinity-disabled scenarios.

Reply:
added!